tests/common/devices/sonic.py

```diff
@@ -642,12 +642,21 @@ def all_critical_process_status(self):
             service_group_process = group_process_results[service]
 
             service_result = service_results[service]
+            # If container is not running, stdout_lines is empty
+            # in this situation, service container status should be false
+            if not service_result['stdout_lines']:
+                service_critical_process['status'] = False
             for line in service_result['stdout_lines']:
                 pname, status, _ = re.split('\s+', line, 2)
                 if status != 'RUNNING':
                     if pname in service_group_process['groups'] or pname in service_group_process['processes']:
                         service_critical_process['exited_critical_process'].append(pname)
                         service_critical_process['status'] = False
+                    # Sometimes, stdout_lines may be error messages but not emtpy
+                    # In this situation, service container status should be false
+                    # We can check status is valid or not
+                    if status not in ('EXITED'):
+                        service_critical_process['status'] = False
```

Comment:
This is a little bit confusing. Can you improve the logic here?

Reply:
Thanks, I will improve it.